node/packages/parser/src/ethereum/yearn.ts

```diff
@@ -16,6 +26,13 @@ export class Parser implements GenericParser {
   depositSigHash = '0x20e8c565'
   withdrawSigHash = '0x00f714ce'
 
+  constructor(args: ParserArgs) {
+    this.provider = args.provider
+    // The only Yearn-supported chain we currently support is mainnet
+    const network = args.network === 'mainnet' ? 1 : undefined // 1 for EthMain (@yfi/sdk/dist/chain.d.ts)
+    this.yearnSdk = network ? new Yearn(network, { provider: this.provider, disableAllowlist: true }) : undefined
```

Comment:
maybe just if condition over double ternary?
```suggestion

    // The only Yearn-supported chain we currently support is mainnet
    if (args.network == 'mainnet') {
      // 1 for EthMain (@yfi/sdk/dist/chain.d.ts)
      this.yearnSdk = new Yearn(1, { provider: this.provider, disableAllowlist: true })
    }
```